### ISSUE TITLE
NVDA doesn't read and track the edit area when entering data in cells in Excel 2013

### ISSUE THREAD

**Issue by btman16** [nvaccess/nvda#6420]
I'm not sure if this applies to Excel 2010 and lower, but when I open Excel 2013, and find a cell, when I start typing, NVDA reads "unknown".
When I used Excel before, and I would start typing, it would say "edit" and then read and track the edit area. This also occurs when you press F2 to edit. When you use the arrow keys to read, NVDA doesn't track and read where I'm at. For instance when I type "payment" into cell A1 the expected result is, when I press F2 to edit, NVDA currently says "unknown" but it should, in the case of payment, say p a y m e ... etc as I move with the arrow keys. Another user of NVDA told me that this was a regression.
When it worked before, I was on Excel 2007.

I am using NVDA 2016.3.

Thank you very much for your time and I look forward to any input you may be able to provide.

Thanks,

Brandon

**Comment by fernando-jose-silva** [comment]
Hello.
With me this problem also occurs in Excel 2013 NVDA does not read the editing cells in Excel editing cells is read.
Using NVDA next 13616 and Excel 2013 and Windows 7.
Nah machine where NVDA reads the editing cell'm using Excel 2016 and Windows NVDA 13616 10 14393.
Follow the log of the machine where NVDA does not read the edition of the cells.
IO - inputCore.InputManager.executeGesture (15:26:03):
Input: kb(laptop):alt+tab
IO - speech.speak (15:26:03):
Speaking [u'Pasta1 - Excel  \xedcone  2 de 8']
IO - speech.speak (15:26:03):
Speaking [u'Pasta1 - Excel']
DEBUGWARNING - NVDAObjects.window.excel.ExcelBase._getDropdown (15:26:03):
Could not locate dropdown list in previous objects
IO - speech.speak (15:26:03):
Speaking [u'Plan1  tabela']
IO - speech.speak (15:26:04):
Speaking [u'A1']
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:04):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 8 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:04):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 10 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:04):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 11 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:04):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 13 and childID 0: [Error -2147467259] Erro não especificado
IO - speech.speak (15:26:04):
Speaking [u'cursor Desconhecido']
IO - inputCore.InputManager.executeGesture (15:26:07):
Input: kb(laptop):f2
DEBUGWARNING - NVDAObjects.window.excel.ExcelBrowseModeTreeInterceptor._get_isAlive (15:26:07):
could not compare sheet names
Traceback (most recent call last):
  File "NVDAObjects\window\excel.pyc", line 444, in _get_isAlive
  File "comtypesMonkeyPatches.pyc", line 35, in new__getattr__
  File "comtypes\client\lazybind.pyc", line 149, in __getattr__
  File "comtypes\automation.pyc", line 664, in _invoke
COMError: (-2147418111, 'A chamada foi rejeitada pelo chamado.', (None, None, None, 0, None))
IO - speech.speak (15:26:07):
Speaking [u'desconhecido']
IO - inputCore.InputManager.executeGesture (15:26:09):
Input: kb(laptop):t
IO - speech._speakSpellingGen (15:26:09):
Speaking character u't'
IO - inputCore.InputManager.executeGesture (15:26:09):
Input: kb(laptop):e
IO - speech._speakSpellingGen (15:26:09):
Speaking character u'e'
IO - inputCore.InputManager.executeGesture (15:26:10):
Input: kb(laptop):s
IO - speech._speakSpellingGen (15:26:10):
Speaking character u's'
IO - inputCore.InputManager.executeGesture (15:26:10):
Input: kb(laptop):t
IO - speech._speakSpellingGen (15:26:10):
Speaking character u't'
IO - inputCore.InputManager.executeGesture (15:26:10):
Input: kb(laptop):e
IO - speech._speakSpellingGen (15:26:10):
Speaking character u'e'
IO - inputCore.InputManager.executeGesture (15:26:11):
Input: kb(laptop):enter
IO - speech.speakTypedCharacters (15:26:11):
typed word: teste
IO - speech.speak (15:26:11):
Speaking [u'teste']
DEBUGWARNING - NVDAObjects.window.excel.ExcelBase._getDropdown (15:26:11):
Could not locate dropdown list in previous objects
IO - speech.speak (15:26:11):
Speaking [u'Plan1  tabela']
IO - speech.speak (15:26:11):
Speaking [u'A2']
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:11):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 76 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:11):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 134 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:11):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 135 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:11):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 136 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:11):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 137 and childID 0: [Error -2147467259] Erro não especificado
IO - inputCore.InputManager.executeGesture (15:26:13):
Input: kb(laptop):upArrow
IO - speech.speak (15:26:13):
Speaking [u'teste  A1']
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:13):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 138 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:13):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 139 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:13):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 140 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:13):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 141 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:13):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 142 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:13):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 143 and childID 0: [Error -2147467259] Erro não especificado
IO - inputCore.InputManager.executeGesture (15:26:14):
Input: kb(laptop):f2
DEBUGWARNING - NVDAObjects.window.excel.ExcelBrowseModeTreeInterceptor._get_isAlive (15:26:14):
could not compare sheet names
Traceback (most recent call last):
  File "NVDAObjects\window\excel.pyc", line 444, in _get_isAlive
  File "comtypesMonkeyPatches.pyc", line 35, in new__getattr__
  File "comtypes\client\lazybind.pyc", line 149, in __getattr__
  File "comtypes\automation.pyc", line 664, in _invoke
COMError: (-2147418111, 'A chamada foi rejeitada pelo chamado.', (None, None, None, 0, None))
IO - speech.speak (15:26:14):
Speaking [u'desconhecido']
IO - inputCore.InputManager.executeGesture (15:26:16):
Input: kb(laptop):control+home
IO - inputCore.InputManager.executeGesture (15:26:17):
Input: kb(laptop):rightArrow
IO - inputCore.InputManager.executeGesture (15:26:17):
Input: kb(laptop):rightArrow
IO - inputCore.InputManager.executeGesture (15:26:21):
Input: kb(laptop):escape
DEBUGWARNING - NVDAObjects.window.excel.ExcelBase._getDropdown (15:26:21):
Could not locate dropdown list in previous objects
IO - speech.speak (15:26:21):
Speaking [u'Plan1  tabela']
IO - speech.speak (15:26:22):
Speaking [u'teste  A1']
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:22):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 258 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:22):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 259 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:22):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 260 and childID 0: [Error -2147467259] Erro não especificado
DEBUGWARNING - IAccessibleHandler.accessibleObjectFromEvent (15:26:22):
oleacc.AccessibleObjectFromEvent with window 460958, objectID 261 and childID 0: [Error -2147467259] Erro não especificado
IO - inputCore.InputManager.executeGesture (15:26:23):
Input: kb(laptop):NVDA+f1
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (15:26:23):
Developer info for navigator object:
name: u'teste'
role: ROLE_TABLECELL
states: 
isFocusable: False
hasFocus: False
Python object: <NVDAObjects.window.excel.ExcelCell object at 0x0B026510>
Python class mro: (<class 'NVDAObjects.window.excel.ExcelCell'>, <class 'NVDAObjects.window.excel.ExcelBase'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: (0, 189, 1600, 649)
value: ''
appModule: <'excel' (appName u'excel', process ID 1576) at address 5cf8e10>
appModule.productName: u'Microsoft Office 2013'
appModule.productVersion: u'15.0.4420.1017'
TextInfo: <class 'NVDAObjects.window.excel.ExcelCellTextInfo'>
windowHandle: 460958
windowClassName: u'EXCEL7'
windowControlID: 0
windowStyle: 1460338688
windowThreadID: 5536
windowText: u'Pasta1'
displayText: u''

**Comment by jcsteh** [comment]
* This works as expected for us on several systems with both Office 2013 and 2016.
* Another user experienced this; see #5384 for details. For this user, the problem went away after reinstalling some components (specifically, the .net 4.5 SDK, though I don't see how this is related).
* Is this a 32 bit or 64 bit version of Office?
* Please try uninstalling and reinstalling Microsoft Office to see if this fixes the problem.

**Comment by jcsteh** [comment]
P2 because this is fairly high impact. This is a problem in Excel on some systems and there's nothing we can do in NVDA to work around this, but it'd be good to have a set of instructions to repair the problem on the system.

**Comment by btman16** [comment]
Hi,

I have no idea about what kind of Office it is. Tech people at the
school installed it, so if I re-installed it I can't do this on my own
in this case.
Plus, several changes were made to the computer which could have
impacted this, for instance one thing was done that makes NVDA take 30
percent of the CPU ane not read task manager.

Thanks,

Brandon

On 10/5/16, James Teh <notifications@github.com> wrote:
> P2 because this is fairly high impact. This is a problem in Excel on some
> systems and there's nothing we can do in NVDA to work around this, but it'd
> be good to have a set of instructions to repair the problem on the system.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/6420#issuecomment-251852595


-- 
“Be what you are. This is the first step towards becoming better than you are.”
– J. C. Hare & A. W. Hare

**Comment by fernando-jose-silva** [comment]
Thank you for feedback.

The machine where NVDA does not read the cell issue was with the net frame work 4.6, uninstalled and reinstalled.
I downloaded the frame work net 4.5 sdk and installed on the machine.
I also asked for the support reinstall the office in 2013 because I could not do this because this machine do not have registration keys and this is a machine company where I work.
And unfortunately even after all this continues with the same problem.
The office is the 2013 32-bit, the machine is running Windows 7 Enterprise 64-bit.

**Comment by Adriani90** [comment]
@btman16, @fernando-jose-silva this should be resolved in NVDA 2019.1 Beta 1. Could you please test if the issue is still occuring? Thanks!

**Comment by fernando-jose-silva** [comment]
I no longer have access to problematic machines.
But in tests on a machine with windows 10 17763.348, nvda alfa 16885 and office 2019 everything works fine.

**Comment by Novalis7747** [comment]
this was solved for me in Office 2013 and NVDA 2018.4.1 and Office 365 with NVDA 2018.4.1 and latest 2019.1 alpha as well. I can't speak for excel versions older than Excel 2013. If I remember well this was already solved for Office 365 in an earlier NVDA versions in 2018...

**Comment by Adriani90** [comment]
Thanks Ralph and Fernando for your promt reaction. Closing as works for me.
